### DISCUSSION TITLE
Detecting when an event video is fully written

### DISCUSSION THREAD

**Discussion by ediguidedog** [blakeblackshear/frigate#5266] [Ask A Question] [answered]
I've been experimenting with resizing the full resolution event video into something much smaller for sending to telegram and the steps pretty much are as below 

```
1 MQTT event received 
2 Event checked for end / person / zones etc.
3 Wait till event end time is at least 5 seconds in the past  
4 Call ffmpeg to resize event video to 320x200 
5 send video via telegram 

```
The bit I'm struggling for consistency is step 3. It seems the end MQTT event is sent before the resulting event MP4 file is actually written to disk and while a simple wait of 5 seconds usually does the trick, sometimes it doesn't and I end up with a truncated event file. Of course I can increase this to 10 or 15 second but then I'm introducing an unnecessarily long wait for most occurrences when the file was ready much earlier

Is there a way of determining when the file has fully written to disk or it it really just a case of polling the file size every second and then proceeding when it stops growing?

Thanks etc.

**Comment by NickM-27** [comment, marked as answer]
This is the best way currently, there is a feature request to add another payload to be sent for when the recording segment has been written to disk

**Reply by ediguidedog** [reply]
Perfect thankyou
